### ISSUE TITLE
kubernetes: PTR issue of some sort

### ISSUE THREAD

**Issue by johnbelamaric** [coredns/coredns#503]
I have CoreDNS running in a test cluster and I am seeing an error on the client side when doing a reverse lookup and one on the server side when looking up something not in the default namespace (kube-dns.kube-system from a pod in default, for example).

Logs:
```
ubuntu@godev:~$ k logs --namespace kube-system coredns-243894951-qzbf1 --follow
2017/02/06 21:17:17 [INFO] Kubernetes middleware configured without a label selector. No label-based filtering will be performed.
.:53
2017/02/06 21:17:17 [INFO] CoreDNS-004
CoreDNS-004
10.2.6.75 - [06/Feb/2017:21:18:44 +0000] "A IN kubernetes.default.svc.cluster.local. udp 54 false 512" NOERROR 70 981.518µs
10.2.6.75 - [06/Feb/2017:21:18:44 +0000] "AAAA IN kubernetes.default.svc.cluster.local. udp 54 false 512" NOERROR 107 206.432µs
10.2.6.75 - [06/Feb/2017:21:18:44 +0000] "MX IN kubernetes.default.svc.cluster.local. udp 54 false 512" NOERROR 107 269.595µs
10.2.6.75 - [06/Feb/2017:21:18:46 +0000] "PTR IN 1.0.3.10.in-addr.arpa. udp 39 false 512" NOERROR 91 474.15µs
10.2.6.75 - [06/Feb/2017:21:23:11 +0000] "PTR IN 1.1.3.10.in-addr.arpa. udp 39 false 512" NXDOMAIN 115 4.210922ms
10.2.6.75 - [06/Feb/2017:21:23:24 +0000] "PTR IN 8.8.8.8.in-addr.arpa. udp 38 false 512" NOERROR 82 244.647573ms
10.2.6.75 - [06/Feb/2017:21:23:53 +0000] "A IN kube-dns.kube-system.default.svc.cluster.local. udp 64 false 512" NXDOMAIN 117 844.953µs
06/Feb/2017:21:23:53 +0000 [ERROR 0 kube-dns.kube-system.default.svc.cluster.local. A] no items found
10.2.6.75 - [06/Feb/2017:21:23:53 +0000] "A IN kube-dns.kube-system.svc.cluster.local. udp 56 false 512" NOERROR 72 702.577µs
10.2.6.75 - [06/Feb/2017:21:23:53 +0000] "AAAA IN kube-dns.kube-system.svc.cluster.local. udp 56 false 512" NOERROR 109 178.441µs
10.2.6.75 - [06/Feb/2017:21:23:53 +0000] "MX IN kube-dns.kube-system.svc.cluster.local. udp 56 false 512" NOERROR 109 301.278µs
10.2.6.75 - [06/Feb/2017:21:23:58 +0000] "PTR IN 10.0.3.10.in-addr.arpa. udp 40 false 512" NOERROR 92 662.006µs
```

From queries:
```
jbelamaric@control:~$ k run -it --image=infoblox/dnstools dnstest18
Waiting for pod default/dnstest18-1012047648-4f0fk to be running, status is Pending, pod ready: false
If you don't see a command prompt, try pressing enter.
/ # host kubernetes
kubernetes.default.svc.cluster.local has address 10.3.0.1
/ # host 10.3.0.1
;; Warning: Message parser reports malformed message packet.
1.0.3.10.in-addr.arpa has no PTR record
/ # host 10.3.1.1
Host 1.1.3.10.in-addr.arpa. not found: 3(NXDOMAIN)
/ # host 8.8.8.8
8.8.8.8.in-addr.arpa domain name pointer google-public-dns-a.google.com.
/ # host kube-dns.kube-system
kube-dns.kube-system.svc.cluster.local has address 10.3.0.10
/ # host 10.3.0.10
;; Warning: Message parser reports malformed message packet.
10.0.3.10.in-addr.arpa has no PTR record
/ #
```

Corefile:
```
    .:53 {
        errors
        log stdout
        health
        kubernetes cluster.local {
          cidrs 10.3.0.0/24
        }
        proxy . /etc/resolv.conf
        cache 30
    }
```

**Comment by chrisohaver** [comment]
I have fixed the PTR issue in PR #506.

For the server side errors (no records, and namespace not exposed), these only show up when errors are displayed (errors middleware).  Should we suppress these from the output always?

**Comment by miekg** [comment]
[ Quoting <notifications@github.com> in "Re: [miekg/coredns] kubernetes: PTR..." ]
>I have fixed the PTR issue in PR #506.
>
>For the server side errors (no records, and namespace not exposed), these only show up when errors are displayed (errors middleware).  Should we suppress these from the output always?

Does that logging add anything for debugging purposes? If not, then delete them.
